cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/Asset.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.metadata;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+import java.util.Objects;
+
+/**
+ * Represents a dataset with a FQN which is a fully-qualified unique identifier for the dataset
```

Comment:
If it is representing dataset, why not call this class `Dataset`? It is a bit strange to use the term `dataset` in the doc but then call the class `Asset`

Reply:
Yeah, I see that we primarily use dataset as the term so will stick to that